cloud/scope/powervs_cluster.go

```diff
@@ -1627,71 +1648,76 @@ func (s *PowerVSClusterScope) ReconcileTransitGateway() (bool, error) {
 		if err != nil {
 			return false, err
 		}
-		requeue, err := s.checkTransitGateway(tg.ID)
+		requeue, _, _, err := s.checkAndUpdateTransitGateway(tg, false)
 		if err != nil {
 			return false, err
 		}
 		return requeue, nil
 	}
 
 	// check transit gateway exist in cloud
-	tgID, requeue, err := s.isTransitGatewayExists()
+	tgID, requeue, powerVSConn, vpcConn, err := s.isTransitGatewayExists()
 	if err != nil {
 		return false, err
 	}
 	if tgID != "" {
-		s.V(3).Info("Transit gateway found in IBM Cloud")
-		s.SetStatus(infrav1beta2.ResourceTypeTransitGateway, infrav1beta2.ResourceReference{ID: &tgID, ControllerCreated: ptr.To(false)})
+		s.SetTransitGatewayStatus(&tgID, ptr.To(false), powerVSConn, vpcConn)
```

Comment:
It won't come to this block on second reconciliation since status would have already generated and try to validate the resource's status in 1st block.